smac/facade/smac_ac_facade.py

```diff
@@ -326,12 +330,33 @@ def __init__(self,
                 **acq_def_kwargs
             )
 
+        if optimize_with_priors:
+            # TODO - could also consider checking for parameter types - If any of the 
+            # prior parameters are in the configspace, we optimize with priors
+
+            # a solid default value for decay_beta - empirically founded
+            default_beta = scenario.ta_run_limit / 10
+            decay_beta = acq_def_kwargs.get('decay_beta', default_beta)          
+            discretize = isinstance(model_instance, RandomForestWithInstances) or isinstance(model_instance, RFRImputator)  
+            acquisition_function_instance = PriorAcquisitionFunction(
+                acquisition_function=acquisition_function_instance, 
+                decay_beta=decay_beta,
+                discretize=discretize,
+                **acq_def_kwargs
+            )
+            uniform_config_space = scenario.cs
```

Comment:
There is a new local search added, which retrieves half of the starting points from the prior and half uniformly. This was added strictly because it proved to be better empirically. If we just go with the default, the local search starting points will be sampled from the prior, which is a bit too biased (particularily if the prior is incorrect).

So, the new local search needs to sample from two different configspaces - the one defined by the user and one where all parameters are uniform.

Reply:
Thanks for your elaboration, makes sense!

Reply:
Still, do we need Line 347 specifically? Because `scenario.cs` at this point is not a uniform space and `uniform_config_space` is not used (?)

Reply:
Agree, that's unnecesary!